### PR TITLE
Preserve data in the browser

### PR DESCRIPTION
**Pull request by NeckerFree**
Use Local storage  to preserve contact form data

Preserve data

### CODE REVIEW

File: js/scriptCode.js [hunk 1]
```diff
@@ -1,9 +1,71 @@
...
+});
+
+window.addEventListener('load', () => {
+  if (storageAvailable('localStorage')) {
+    if (localStorage.getItem('contactData')) {
+      setContactForm();
+    }
+  } else {
+    alert('Local Storage not avalailable');
+  }
```
Comment (BenMukebo) [inline js/scriptCode.js]: - [x] Kindly make the contact form information persistent to page reloads, currently, if users don't submit after inputting the data goes away
 
- [x] Please do not use alert() in your codebase, I would suggest you remove them and try other methods.

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi **@NeckerFree,**

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Good commit messages ✔️ 

 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [x] Please, Write a meaningful title for your pull request and always add a short summary of changes introduced in the pull request. you can check the  [Professional repo - basic rules](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏 💻  

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/scriptCode.js (hunk 1 above)
Comment (medaminedev66) [reply js/scriptCode.js]: - [ ] Following the project requirement, after filling the form, **and not submitting it**, the entered values should be saved in the browser (localStorage). If the user reloads the page or closes the browser and opens it again, the form inputs should be auto-populated with the values entered in the previous visit to the page.
- [ ] As requested by the previous reviewer, please remove the alert() from your code to follow the[ JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: js/scriptCode.js (hunk 1 above)
Comment (NeckerFree) [reply js/scriptCode.js]: Sorry, I missed the push

**Review by keroloussamy — APPROVED** [review]
Hi @NeckerFree, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**